xml/cl.xml

```diff
@@ -1648,33 +1723,57 @@ server's OpenCL/api-docs repository.
         <enum value="0x120C"        name="CL_COMMAND_SVM_MAP"/>
         <enum value="0x120D"        name="CL_COMMAND_SVM_UNMAP"/>
         <enum value="0x120E"        name="CL_COMMAND_SVM_MIGRATE_MEM"/>
-            <unused start="0x120F" end="0x121F" comment="Reserved for cl_command_types"/>
+            <unused start="0x120F" end="0x121F" comment="Reserved for cl_command_type"/>
+    </enums>
+
+    <enums start="0x1220" end="0x127F" name="cl_buffer_create_type" vendor="Khronos">
         <enum value="0x1220"        name="CL_BUFFER_CREATE_TYPE_REGION"/>
             <unused start="0x1221" end="0x127F" comment="Reserved for cl_buffer_create_type"/>
+    </enums>
+
+    <enums start="0x1280" end="0x128F" name="cl_profiling_info" vendor="Khronos">
         <enum value="0x1280"        name="CL_PROFILING_COMMAND_QUEUED"/>
         <enum value="0x1281"        name="CL_PROFILING_COMMAND_SUBMIT"/>
         <enum value="0x1282"        name="CL_PROFILING_COMMAND_START"/>
         <enum value="0x1283"        name="CL_PROFILING_COMMAND_END"/>
         <enum value="0x1284"        name="CL_PROFILING_COMMAND_COMPLETE"/>
             <unused start="0x1285" end="0x128F" comment="Reserved for cl_profiling_info"/>
+    </enums>
+
+    <enums start="0x1290" end="0x1292" name="enums.1290" vendor="Khronos">
             <unused start="0x1290" end="0x1291" comment="Reserved for MR179"/>
         <enum value="0x1292"        name="Reserved for PR750"/>
+    </enums>
+
+    <enums start="0x1293" end="0x1293" name="cl_command_buffer_properties_khr" vendor="Khronos">
         <enum value="0x1293"             name="CL_COMMAND_BUFFER_FLAGS_KHR"/>
+    </enums>
+
+    <enums start="0x1294" end="0x12A7" name="cl_command_buffer_info_khr" vendor="Khronos">
         <enum value="0x1294"             name="CL_COMMAND_BUFFER_QUEUES_KHR"/>
         <enum value="0x1295"             name="CL_COMMAND_BUFFER_NUM_QUEUES_KHR"/>
         <enum value="0x1296"             name="CL_COMMAND_BUFFER_REFERENCE_COUNT_KHR"/>
         <enum value="0x1297"             name="CL_COMMAND_BUFFER_STATE_KHR"/>
         <enum value="0x1298"             name="CL_COMMAND_BUFFER_PROPERTIES_ARRAY_KHR"/>
             <unused start="0x1299" end="0x12A7" comment="Used by future command-buffer extensions"/>
+    </enums>
+
+    <enums start="0x12A8" end="0x12A8" name="cl_command_type" vendor="Khronos">
         <enum value="0x12A8"             name="CL_COMMAND_COMMAND_BUFFER_KHR"/>
+    </enums>
+
+    <enums start="0x12A9" end="0x12B1" name="cl_device_info" vendor="Khronos">
```

Comment:
This change cause one of the scripts ("gencl") executed by the spec Makefile to start to complain:

```sh
scripts/gencl.py    -registry xml/cl.xml -o /home/bashbaug/git/OpenCL-Docs/generated/api apiinc
WARNING: Attempt to redefine cl_command_type (this should not happen)
WARNING: Attempt to redefine cl_device_info (this should not happen)
WARNING: Attempt to redefine cl_kernel_sub_group_info (this should not happen)
```

I'm not sure if the best way to fix this is in the script or in the XML file, but I bet @oddhack knows...

Reply:
Probably best to suppress the warning in the script *if* there is no 'type' attribute on the \<enums> tag. If there's a 'type' attribute, then the \<enums> are defining a derived type with that name, rather than the name simply being a label on a bunch of constants. If you want I'll propose an MR to that effect.